libraries/AP_HAL/examples/RCInput/RCInput.cpp

```diff
@@ -34,7 +34,7 @@ void loop(void)
         }
     }
     if (changed) {
-        for (uint8_t i=0; i<max_channels; i++) {
+        for (uint8_t i=0; i<=max_channels; i++) {
```

Comment:
This PR #4417 solve the problem and confusion, please check.

@dAdil  are right, the code goes wrong with arrays, i implemented according and suggestions from @lucasdemarchi  and @guludo.

Greetings.
Hiroshi.